spec/javascripts/unit/core/pusher_spec.js

```diff
@@ -310,13 +310,31 @@ describe("Pusher", function() {
     });
 
     describe("#unsubscribe", function() {
-      it("should unsubscribe the channel", function() {
+      it("should unsubscribe the channel if subscription is not pending", function() {
         var channel = pusher.subscribe("yyy");
         expect(channel.unsubscribe).not.toHaveBeenCalled();
 
         pusher.unsubscribe("yyy");
         expect(channel.unsubscribe).toHaveBeenCalled();
       });
+
+      it("should remove the channel from .channels if subscription is not pending", function () {
+        var channel = pusher.subscribe("yyy");
+        expect(pusher.channel("yyy")).toBe(channel);
+
+        pusher.unsubscribe("yyy");
+        expect(pusher.channel("yyy")).toBe(undefined);
+      });
+
+      it("should delay unsubscription if the subscription is pending", function () {
+        var channel = pusher.subscribe("yyy");
+        channel.subscriptionPending = true;
```

Comment:
The channel object in the pusher_spec tests is just a mock, so it doesn't do anything except track when it has been called :) The real channel object does indeed set `subscriptionPending` when you call `.subscribe`. This test just makes sure that when the subscription is still pending, the `pusher` object doesn't try to call `.unsubscribe()` on the channel, and instead calls `.cancelSubscripiton()`.